### PR TITLE
VersionClient supports getApiFeatures

### PR DESCRIPTION
**Pull request by adriaanm-da**
This exposes the `features` part of the version service's
`GetLedgerApiVersionResponse` protobuf.

Map the protobuf to user-friendly subclasses of `Feature`,
defined in `c.d.ledger.api.domain`.

This was originally developed (and reviewed) as part of #12187

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala [hunk 1]
```diff
@@ -22,4 +35,87 @@ private[daml] final class VersionClient(service: VersionServiceStub) {
...
+      .map(_.features.toList.flatMap(VersionClient.fromProto))
+}
+
+private[daml] object VersionClient {
+  // see also com.daml.platform.apiserver.services.ApiVersionService.featuresDescriptor
+  def fromProto(featuresDescriptor: FeaturesDescriptor): Seq[Feature] =
+    featuresDescriptor match {
+      case FeaturesDescriptor(userManagement, experimentalFeatures) =>
+        (userManagement.toList map (_ => Feature.UserManagement)) ++
+          (experimentalFeatures.toList flatMap fromProto)
```
Comment (meiersi-da) [inline ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala:57]: @adriaanm-da : do you need to use these experimental features? They are essentially internal features that we don't want to expose to customers. So, I'm not sure whether we should expose them in the official client. At least not at the same level as the official features. Do you see a good way to provide this distinction at the API level?

Comment (meiersi-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala:57]: I guess my intuitive preference is to not expose them at all.

Comment (adriaanm-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala:57]: I only needed `userManagement`, and then just went for exhaustiveness for no good reason other than to appease scalac :-) 

Happy to lock the API down more -- I hadn't considered the customer angle. How do we make a decision? Who else should I loop in?

Comment (meiersi-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala:57]: In general, we can always open up our API's more. Locking them down is the hard part.

As for the decision, I'd like to know what @stefanobaghino-da and @gerolf-da think, and then make a decision based on our input.

My vote currently goes to "do not expose experimental features".

**Comment by adriaanm-da** [comment]
After digging in the logs, all I can see is likely spurious failures. Should I rerun?

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala [hunk 2]
```diff
@@ -22,4 +23,24 @@ private[daml] final class VersionClient(service: VersionServiceStub) {
...
+        new GetLedgerApiVersionRequest(ledgerIdToUse.unwrap)
+      )
+      .map(_.features.toList.flatMap(VersionClient.fromProto))
+}
+
+private[daml] object VersionClient {
+  // see also com.daml.platform.apiserver.services.ApiVersionService.featuresDescriptor
+  def fromProto(featuresDescriptor: FeaturesDescriptor): Seq[Feature] =
+    featuresDescriptor match {
+      case FeaturesDescriptor(userManagement, _ /* TODO: expose experimentalFeatures */ ) =>
```
Comment (meiersi-da) [inline ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala:43]:
```suggestion
      // Note that we do not expose experimental features here, as they are used for internal testing only
      // and do not have backwards compatibility guarantees. (They should probably be named 'internalFeatures' ;-)
      case FeaturesDescriptor(userManagement, _ ) =>
```

**Review by meiersi-da — APPROVED** [review]
Thank you @adriaanm-da . One small suggestion, and then it's ready to merge! :pray:

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala (hunk 2 above)
Comment (adriaanm-da) [reply ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/version/withoutledgerid/VersionClient.scala]: Thanks for the suggestion! Added to the PR.

**Review by stefanobaghino-da — APPROVED** [review]

**Pull request merged**